### ISSUE TITLE
Future / Documentation / Roadmap / Stuff of that sort

### ISSUE THREAD

**Issue by tobimensch** [tmbdev/clstm#26]
Hi,

I was working on a GUI for scanning PDF documents and I was looking
for a way to have correct OCR under Linux and to sandwich the text into the generated PDFs.

Obviously the state of open source OCR isn't stellar, everything I tried wasn't exactly a breeze
to setup and most importantly the actual detection was quite bad for example with tesselact and
all the other known open source projects.

Then I read about ocropy/ocropus/clstm, and finally was relieved that someone was at last working from scratch on an innovative modern approach, that was actually being actively developed and some of the results I saw in one online tutorial where a guy modified the model for typewriter fonts and it worked rather good were promising.

So I got ocropy to work on Fedora and tried it with the standard model on a simple plain PDF document I had scanned. The results were just as terrible as with the other open source OCRs, if not worse, although it really is a high resolution scan with normal looking fonts. I felt disappointed to be honest.

From the documentation/README.md and so on I didn't get a lot of information about where the project is going and where it might need some help. The basic concepts are explained well, but I didn't get a good picture of where the problems of the project are and what is done to solve them.
It seems "the model" is just a file that was uploaded sometime ago, but doesn't get actively maintained and improved currently.

I  would be interesting to know what the obstacles currently are for supporting more fonts/fontstyles and so on. Is there any plan for what the next milestones should be? Is there any guide how people can best help the project to fix bugs, add features, refactor code, improve models and so on.

I'm writing this in this project, because I really see it as the same project basically.

To summarize:
- It would be nice if you kept a detailed TODO list for clstm and ocropy
- It would be nice to have a Roadmap for both those projects as a whole where
  important milestones are defined
- The documentation should explain how people can help with development,
  especially how models can be created and improved upon, so that they can
  eventually be included as "official" models. Do you for example require some
  example PDFs from people that currently don't work well in order for you to
  improve future models?

I'm sorry, if this doesn't fit into "issues". Feel free to remove or move this, I just
wanted to express that what's going on with ocropy/clstm development seems
like a blackbox from the outside.

**Comment by tmbdev** [comment]
> So I got ocropy to work on Fedora and tried it with the standard model on a simple plain PDF document I had scanned. The results were just as terrible as with the other open source OCRs, if not worse, although it really is a high resolution scan with normal looking fonts. I felt disappointed to be honest.

Error rates even on degraded documents are generally below 1% character error. If you're getting worse than that on clean documents, there's probably something wrong. 

To diagnose the source of the error, please provide extracted text lines as samples; you can attach them to this issue.

**Comment by tmbdev** [comment]
> It would be nice if you kept a detailed TODO list for clstm and ocropy. It would be nice to have a Roadmap for both those projects as a whole where important milestones are defined.

Unfortunately, the development of ocropy and clstm is largely driven by a lot of internal project timelines and other considerations. I'll try to add some comments to the READMEs, though.

**Comment by anupamaray** [comment]
does clstm work on windows?

**Comment by tmbdev** [comment]
That could mean several things. Porting to MSVC++ is probably a modest amount of work. It should compile with Cygwin. I'd recommend just running Virtualbox or Docker, though.,
